Relax the prism test parser conditions

Earlopain opened the pull request:
### Motivation / Background

Closes #51956

### Detail

Checking explicitly against `test` break extensions that provide their own methods to generate tests, like `minitest-spec-rails` or `minitest-rails`. The ripper parser had no such checks, so I just dropped those.

cc @kddnewton

### Additional information

Needs backporting to 7.2. Maybe changelog?

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

kddnewton reviewed (commented):
Ahh I missed that it didn't check that. Thanks!

rafaelfranca commented on railties/test/test_unit/test_parser_test.rb:
```diff
@@ -38,6 +38,19 @@ def test_oneline; assert true; end
     assert_not false
   }
+
+  # Emulate minitest/spec. Check that extensions can provide aliases
```
Can we use something else to test this? We really don't want to make like we support minitest/spec or any syntax that looks like rspec.

Earlopain replied:
Yeah, sure. I made it look more generic

Earlopain committed:
Relax the prism test parser conditions

Checking explicitly against `test` break extensions that provide their
own methods to generate tests, like `minitest-spec-rails` or `minitest-rails`.

Fixes #51956
